### ISSUE TITLE
Add  mode in trips and vehicles objects

### ISSUE THREAD

**Issue by pierre-bouffort** [openmobilityfoundation/mobility-data-specification#887]
In the present data structure, the `trips` and `vehicles` objects have some fields that depend on the Mode (`trip_attribute`, `journey_attribute`, `fare_attributes`, ...). 
But currently, the `trips` and `vehicles` do not have a `mode` field.  Therefore, to determine the mode of a trip for example, we have to retro-engineer it from the  `trip_attribute`, `journey_attribute`, `fare_attributes`, ... fields. 

=> This can cause some issues when trying to validate and store the data correctly.

**Comment by jiffyclub** [comment]
The expected mode of a data object can be determined from the `provider_id`. [Per the spec](https://github.com/openmobilityfoundation/mobility-data-specification/tree/2.0.0/modes#provider-id) "a provider implementing more than one mode shall [register](https://github.com/openmobilityfoundation/mobility-data-specification/blob/2.0.0/README.md#providers-using-mds) a unique provider_id for each mode." The [`providers.csv` file](https://github.com/openmobilityfoundation/mobility-data-specification/blob/2.0.0/providers.csv) contains the mode associated with each provider.

**Comment by kevin-courbet-blue** [comment]
Good point. Although this means that, in the context of processing a given document (trip, or vehicle registration), the document is not stand-alone, ie some details are hidden away until one cross references the providers table. While technically it is possible to cross-reference the providers table at runtime, I think it is safer/easier/more explicit if the property is part of the document itself. It is easier to implement because you know what shape the document is supposed to be just by looking at it. Although cross-referencing the providers table isn't too hard, it's still something extra to do that could be avoided.  @pierre-bouffort  fyi

**Comment by jiffyclub** [comment]
I vaguely remember a conversation in the development of MDS where we explicitly decided not to include mode in data objects since it's duplicate information, both from the provider ID and because there's a 1:1 mapping between a mode and download URL. You have to add mode to pretty much every single data object in MDS if you go down this route, but I see the appeal.

**Comment by schnuerle** [comment]
Yes @jiffyclub is correct. We had a few discussions about this on GitHub and in working groups. The consensus was to store the mode info only in the providers.csv file, and not duplicate mode info again in the data, to avoid any potential conflicting mode info and not know which is the authoritative source.

**Comment by kevin-courbet-blue** [comment]
It just occurred to me that, at least in the case of OAuth2 (which implementation uses), one could add a `mode` scope to the JWT (eg. `providers/car_share`), so if implementing this way, a round-trip to the providers dataset is not necessary (the mode will be available for each request).

I still think it would be beneficial for data structures to be "stand-alone" for easier processing down the road, but at least this isn't mandatory anymore, more like a strong recommendation.

@pierre-bouffort  fyi

**Comment by schnuerle** [comment]
I think we can close this issue, correct? The consensus is to use the providers.csv file to determine the mode and is the single source of truth.
